# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1759859876 -7200
# Node ID cc423f768e9e72c7568e8c174eb011c9e895a4c2
# Parent  b9f0a38764d12d5fe0d3a2a207d9148d06be15df
Bug 1992351 - Improve fix for Bug 1912195. r=darktrojan,gds

Since the message database for the folder is closed and re-created in
`folderPane.rebuildFolderSummary`, performing a compress commit immediately
afterwards shouldn't have any effect whatsoever. (For local folders this
doesn't even work, since parsing hasn't finished at this time)

The actual fix in `nsMsgDBFolder::HandleAutoCompactEvent` should be performing
a compress (not a large) commit when compacting the folder isn't necessary.

Differential Revision: https://phabricator.services.mozilla.com/D267666

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3590,9 +3590,8 @@ var folderPane = {
       // In a failure, proceed anyway since we're dealing with problems
       folder.ForceDBClosed();
     }
     folder.updateFolder(top.msgWindow);
-    folder.msgDatabase.commit(Ci.nsMsgDBCommitType.kCompressCommit);
   },
 
   /**
    * Opens the dialog to edit the properties for a folder
diff --git a/mailnews/base/src/nsMsgDBFolder.cpp b/mailnews/base/src/nsMsgDBFolder.cpp
--- a/mailnews/base/src/nsMsgDBFolder.cpp
+++ b/mailnews/base/src/nsMsgDBFolder.cpp
@@ -1699,9 +1699,9 @@ nsresult nsMsgDBFolder::HandleAutoCompac
       }
     } else {
       // Ensure commit happens regularly, even if offline stores are disabled.
       if (mDatabase) {
-        mDatabase->Commit(nsMsgDBCommitType::kLargeCommit);
+        mDatabase->Commit(nsMsgDBCommitType::kCompressCommit);
       }
     }
   }
   return rv;
